introduction-to-sophia.md

```diff
@@ -0,0 +1,134 @@
+# TUTORIAL: Introduction to Sophia
+
+## Tutorial overview
+This tutorial aims to provide an introduction to Sophia's syntax and some of its features with a "Hello, world"-esque smart contract. Following this tutorial should give the reader an idea of the basic practices when writing Sophia smart contracts. The contract that we will be writing will enable its callers to send a greeting message to an oracle and reply to those messages on behalf of the oracle. Ideally, readers should have a basic understanding of how oracles work, if not, feel free to read [this document](https://github.com/aeternity/protocol/blob/master/oracles/oracles.md). Let's get into it!
+
+## Contract state and initialization
+While Sophia is a functional language, contracts (although optional) can hold mutable state. For our example, we will be storing a reference to the oracle that will handle messages and send out responses and all of the greeting messages that have been sent its way. Let's begin by defining our contract body, its state and the `init` function that will set the initial state of our contract when the `Create` transaction is executed.
+
+```
+contract Greeter =
```

Comment:
Where should I put this? Reference a tutorial with introduction on how to deploy and call it